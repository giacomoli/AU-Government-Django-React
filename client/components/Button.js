import React, { Fragment } from 'react'
import PropTypes from 'prop-types'

const Button = ({ loading, className, icon, onClick, title, type, style, disabled }) => {
  let content

  if (loading && icon) {
    content = <Fragment><i className="fa fa-circle-o-notch fa-spin fa-fw" aria-hidden="true" />{title}</Fragment>
  } else if (loading && !icon) {
    content = <i className="fa fa-circle-o-notch fa-spin fa-fw" aria-hidden="true" />
  } else if (icon) {
    content = <Fragment><i className={icon} aria-hidden="true" /> {title}</Fragment>
  } else {
    content = title
  }

  return (
    <button
      onClick={onClick}
      className={className}
      disabled={loading || disabled}
      type={type}
      style={style}
    >
      {content}
    </button>
  )
}

Button.propTypes = {
  loading: PropTypes.bool,
  className: PropTypes.string,
  title: PropTypes.string,
  type: PropTypes.string,
  icon: PropTypes.string, // eslint-disable-line
  style: PropTypes.object, // eslint-disable-line
}

Button.defaultProps = {
  loading: false,
  className: '',
  type: 'button',
}

export default Button
