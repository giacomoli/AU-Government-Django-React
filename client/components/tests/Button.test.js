import React from 'react'
import renderer from 'react-test-renderer'
import Button from '../Button'

/* eslint-disable */
describe('Button', () => {

  it('should render Button with some props', () => {
    const wrapper = renderer.create(
        <Button
          className="btn btn-success"
          onClick={jest.fn()}
          title="Text Button"
          type='button'
          style={{color: 'red'}}
          loading={false}
          disabled
        />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render Button with icon props', () => {
    const wrapper = renderer.create(
        <Button
          className="btn btn-success"
          onClick={jest.fn()}
          title="Text Button"
          type='button'
          icon="fa-plus"
          style={{color: 'red'}}
          loading={false}
          disabled
        />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('should render Button with icon and loading props', () => {
    const wrapper = renderer.create(
        <Button
          className="btn btn-success"
          onClick={jest.fn()}
          title="Text Button"
          type='button'
          icon="fa-plus"
          style={{color: 'red'}}
          loading={true}
          disabled
        />
    )
    expect(wrapper).toMatchSnapshot()
  })


  it('should render Button with loading props', () => {
    const wrapper = renderer.create(
        <Button
          className="btn btn-success"
          onClick={jest.fn()}
          title="Text Button"
          type='button'
          style={{color: 'red'}}
          loading={true}
          disabled
        />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
