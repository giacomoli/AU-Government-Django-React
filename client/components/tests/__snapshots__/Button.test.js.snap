// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render Button with icon and loading props 1`] = `
<button
  className="btn btn-success"
  disabled={true}
  onClick={[MockFunction]}
  style={
    Object {
      "color": "red",
    }
  }
  type="button"
>
  <i
    aria-hidden="true"
    className="fa fa-circle-o-notch fa-spin fa-fw"
  />
  Text Button
</button>
`;

exports[`Button should render Button with icon props 1`] = `
<button
  className="btn btn-success"
  disabled={true}
  onClick={[MockFunction]}
  style={
    Object {
      "color": "red",
    }
  }
  type="button"
>
  <i
    aria-hidden="true"
    className="fa-plus"
  />
   
  Text Button
</button>
`;

exports[`Button should render Button with loading props 1`] = `
<button
  className="btn btn-success"
  disabled={true}
  onClick={[MockFunction]}
  style={
    Object {
      "color": "red",
    }
  }
  type="button"
>
  <i
    aria-hidden="true"
    className="fa fa-circle-o-notch fa-spin fa-fw"
  />
</button>
`;

exports[`Button should render Button with some props 1`] = `
<button
  className="btn btn-success"
  disabled={true}
  onClick={[MockFunction]}
  style={
    Object {
      "color": "red",
    }
  }
  type="button"
>
  Text Button
</button>
`;
